src/System.Private.CoreLib/shared/System/Runtime/Intrinsics/X86/Enums.cs

```diff
@@ -167,23 +168,122 @@ public enum FloatComparisonMode : byte
         TrueUnorderedSignaling = 31,
     }
 
+    [Flags]
     public enum StringComparisonMode : byte {
+        /// <summary>
+        /// _SIDD_CMP_EQUAL_ANY
+        /// </summary>
+        EqualAny = 0x00,
+
+        /// <summary>
+        /// _SIDD_CMP_RANGES
+        /// </summary>
+        Ranges = 0x04,
+
+        /// <summary>
+        /// _SIDD_CMP_EQUAL_EACH
+        /// </summary>
+        EqualEach = 0x08,
+
+        /// <summary>
+        /// _SIDD_CMP_EQUAL_ORDERED
+        /// </summary>
+        EqualOrdered = 0x0c,
+
+        /// <summary>
+        /// _SIDD_NEGATIVE_POLARITY
+        /// </summary>
+        NegativeResult = 0x10,
```

Comment:
The architecture manual refers to this as `NegativePolarity`. We should likely also explicitly expose `PositivePolarity`

Reply:
Don't you mean adding a `PositiveResult`? 
I do not think so... That is the default semantics, so `PositiveResult` will have value `0` that makes overlapped values again.

Reply:
> That is the default semantics, so PositiveResult will have value 0 that makes overlapped values again.

From the architecture manual, we actually have a single control byte which is split into multiple sections: `Source Data Format` in bits 0 and 1, `Aggregation Operation` in bits 2 and 3, `Polarity` in bits 4 and 5, and `Output Selection` in bit 6 (bits 7 and 8 are undefined and should be 0)..

Of those, it looks like:
* `Source Data Format` is encoded as part of the signature (the size and sign of the parameters it takes)
* `Aggregation Operation` and `Polarity` are the same for `index`, `mask`, and `bool` modes
  * this PR currently combines them in the `StringComparisonMode` enum
* `Output Selection` behavior differs between `mask` and `index` and doesn't apply to the `bool` mode.
  * It is also currently combined into the `StringComparisonMode` enums for `index`/`mask`.

We should probably just split this into 4 total enums:
* a single `Aggregation Operation` enum (4 values)
* a single `Polarity` enum (4 values)
* two `Output Selection` enums
  * one enum for `index` (2 values)
  * one enum for `mask` (2 values)

Then, the `bool` comparison mode would take `Aggregation Operation` and `Polarity`
The `index` comparison mode would take `Aggregation Operation`, `Polarity`, and `index Output Selection`
The `mask` comparison mode would take `Aggregation Operation`, `Polarity`, and `mask Output Selection`

@CarolEidt, @eerhardt, @terrajobst. Opinions?